ui/StatusQ/src/keychain_osx.mm

```diff
@@ -91,12 +91,14 @@
     return Keychain::StatusSuccess;
 }
 
-void Keychain::requestSaveCredential(const QString &account, const QString &password)
+void Keychain::requestSaveCredential(const QString &reason, const QString &account, const QString &password)
 {
     if (m_future.isRunning()) {
         return;
     }
 
+    m_reason = reason;
```

Comment:
I guess we should pass the reason as argument to the `saveCredential` (and others) as well then?
Same as we pass all other args.